### PR TITLE
Improve visual appearance of ProcessingOverlay

### PR DESCRIPTION
**Pull request by peppy**
Previously it would dim a full area, which could sometimes look very bad (multiplayer lounge room display). Now a dim target can be chosen, allowing only dimming content when more appropriate.

image

Also adds a test scene for this component.

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/ProcessingOverlay.cs [hunk 1]
```diff
@@ -28,29 +35,54 @@ private void load()
...
+            dimTarget?.FadeColour(Color4.White, transition_duration, Easing.OutQuint);
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (State.Value == Visibility.Visible)
+            {
+                // ensure we don't leave the targetin a bad state.
```
Comment (bdach) [inline osu.Game/Graphics/UserInterface/ProcessingOverlay.cs]: classic comment nitpick
```suggestion
                // ensure we don't leave the target in a bad state.
```

**Review by bdach — COMMENTED** [review]
That's a horrible diff for reading, but it's github's fault. It doesn't fare well with indent changes, Rider was much better. Code looks good save for a classic annoying comment suggestion.

As a general note I don't know how I feel about the various loading animations we have in the game on a design language level. We've got a couple of slightly differing variants (this - spinner in a solid box, `DimmedLoadingLayer` - spinner in front of a, well, dimmed layer) and I'm not feeling the consistency too much.

That's something that can be iterated on I guess, and also probably requires design feedback from a designer (which I am not, and my attempts at summoning one have had little effect up to now 😛)

File: osu.Game/Graphics/UserInterface/ProcessingOverlay.cs [hunk 2]
```diff
@@ -28,29 +35,54 @@ private void load()
...
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (State.Value == Visibility.Visible)
+            {
+                // ensure we don't leave the target in a bad state.
+                dimTarget?.FadeColour(Color4.White, transition_duration);
```
Comment (smoogipoo) [inline osu.Game/Graphics/UserInterface/ProcessingOverlay.cs]: Should be `Easing.OutQuint` no?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Graphics/UserInterface/ProcessingOverlay.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Graphics/UserInterface/ProcessingOverlay.cs]: ah, good point. i though this wasn't important as we are already disposed but that logic is back-to-front

**Comment by peppy** [comment]
@bdach yeah, we are still waiting on a more custom loading animation, but for the time being i *think* we only have two display variants. This one should probably be preferred going forward.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**